src/compiler/path.ts

```diff
@@ -547,6 +547,19 @@ namespace ts {
 
     export function normalizePath(path: string): string {
         path = normalizeSlashes(path);
+        // Most paths don't require normalization
+        if (!relativePathSegmentRegExp.test(path)) {
+            return path;
+        }
+        // Some paths only require cleanup of `/./`
+        const simplified = path.replace(/\/\.\//g, "/");

```

Comment:
Wonder if it would be better to clean up leading `./` on paths that are getting concatenated. I suppose sometimes people accidentally do things like `.././` though.

Reply:
Anecdotally, it looks like the arguments are usually absolute paths, so leading `./` is rare.

Reply:
Eh, why not?